programs/redstone-sol/src/instructions/process_redstone_payload.rs

```diff
@@ -38,74 +35,42 @@ pub fn process_redstone_payload(
         block_timestamp: Clock::get()?.unix_timestamp as u64 * 1000,
-        config_account: &ctx.accounts.config_account,
+        signer_count_threshold: ctx
+            .accounts
+            .config_account
+            .signer_count_threshold,
+        signers,
+        feed_ids: vec![feed_id],
     };
 
-    redstone::verify_redstone_marker(&payload)?;
+    let processed_payload = process_payload(config, payload);
 
-    let mut payload = payload;
-    let payload = redstone::parse_raw_payload(&mut payload)?;
-
-    redstone::verify_data_packages(&payload, &config)?;
-
-    #[cfg(feature = "dev")]
+    if ctx.accounts.price_account.timestamp >= processed_payload.min_timestamp
     {
-        msg!(
-            "Payload processed successfully: {}",
-            payload.data_packages.len()
-        );
-        for package in &payload.data_packages {
-            msg!(
-                "Package signer: 0x{}",
-                bytes_to_hex(&package.signer_address)
-            );
-            for data_point in &package.data_points {
-                msg!(
-                    "Data point: {} {}",
-                    u256_to_string(&data_point.feed_id),
-                    u256_to_num_string(&data_point.value),
-                );
-            }
-        }
-    }
-
-    let mut values: Vec<U256> = Vec::new();
-
-    let package_timestamp = payload.data_packages[0].timestamp;
-    for package in &payload.data_packages {
-        if package_timestamp != package.timestamp {
-            return Err(RedstoneError::TimestampMismatch.into());
-        }
-        for data_point in &package.data_points {
-            if feed_id != data_point.feed_id {
-                return Err(RedstoneError::UnsupportedFeedId.into());
-            }
-            values.push(U256::from_bytes_be(&data_point.value));
-        }
-    }
-    if ctx.accounts.price_account.timestamp >= package_timestamp {
         return Err(RedstoneError::TimestampTooOld.into());
     }
 
-    let median_value = median(&values);
-    if let Some(median_value) = &median_value {
-        ctx.accounts.price_account.value = median_value.to_bytes_be();
-    } else {
-        return Err(RedstoneError::MedianCalculationError.into());
-    }
-
-    ctx.accounts.price_account.timestamp = package_timestamp;
-    ctx.accounts.price_account.feed_id = feed_id;
-    ctx.accounts.price_account.write_timestamp = config.block_timestamp;
+    let price = processed_payload.values[0];
+    ctx.accounts.price_account.value = price.to_big_endian();
+    ctx.accounts.price_account.timestamp = processed_payload.min_timestamp;
+    ctx.accounts.price_account.feed_id = feed_id.0;
 
     msg!(
-        "{} {}: {}",
+        "{} {:?}: {:?}",
         ctx.accounts.price_account.timestamp,
-        u256_to_string(&feed_id),
-        u256_to_num_string(&ctx.accounts.price_account.value)
+        feed_id,               // as_hex_strcostly
+        price.to_big_endian(), // conversion to string costly
```

Comment:
we probably could use `print_debug` here instead of msg!

Reply:
as_hex_str[ ]costly

Reply:
added new function for that, print-debug was not suited for inner msg! macro call